File: backup/Twitch/app/src/main/java/com/exact/twitch/ui/fragment/TopPagerAdapter.java
package com.exact.twitch.ui.fragment;

import android.content.Context;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import com.exact.twitch.ui.clips.ClipsFragment;
import com.exact.twitch.ui.pagers.MediaPagerAdapter;
import com.exact.twitch.ui.streams.StreamsFragment;
import com.exact.twitch.ui.videos.TopVideosFragment;

public class TopPagerAdapter extends MediaPagerAdapter {

    public TopPagerAdapter(Context context, FragmentManager fm) {
        super(context, fm);
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new StreamsFragment();
            case 1:
                return new TopVideosFragment();
            case 2:
                return new ClipsFragment();
            default:
                return null;
        }
    }
}

File: app/src/main/java/com/exact/twitch/ui/pagers/TopPagerAdapter.kt
package com.exact.twitch.ui.pagers

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import com.exact.twitch.ui.clips.ClipsFragment
import com.exact.twitch.ui.pagers.MediaPagerAdapter
import com.exact.twitch.ui.streams.StreamsFragment
import com.exact.twitch.ui.videos.TopVideosFragment

class TopPagerAdapter(context: Context, fm: androidx.fragment.app.FragmentManager) : MediaPagerAdapter(context, fm) {

    override fun getItem(position: Int): androidx.fragment.app.Fragment? {
        return when (position) {
            0 -> StreamsFragment()
            1 -> TopVideosFragment()
            2 -> ClipsFragment()
            else -> null
        }
    }
}
